releasenotes/notes/upgrade-pulse-builder-and-rzx-builder-033ac8ad8ad2a192.yaml

```diff
@@ -0,0 +1,26 @@
+---
+features:
+  - |
+    New pulse directive :class:`~qiskit.pulse.instructions.AreaBarrier` has been added.
+    This instruction is QPY compatible. This directive behaves almost identically to
+    delay instruction, but will be removed before execution.
+    This directive is intended to be used internally within the pulse builder
+    and helps :class:`.ScheduleBlock` with representing instructions with
+    absolute time intervals. This allows the pulse builder to convert
+    :class:`Schedule` into :class:`ScheduleBlock`, rather than wrapping with call instruction.
+  - |
+    QPY dump and load now support :class:`~qiskit.pulse.instructions.AreaBarrier` instruction.
+upgrade:
+  - |
+    The behavior of the pulse builder when a :class:`.Schedule` is called
+    has been upgraded. Called schedules are internally converted into
+    :class:`.ScheduleBlock` representation and now reference mechanism is
+    always applied rather than appending the schedules wrapped by
+    the :class:`~qiskit.pulse.instructions.Call` instruction.
```

Comment:
IIRC This conversion assumes a scheduling method that is not in the user's hands. Is this okay?

Reply:
This is technically not scheduling. `Schedule` is the list of instructions which is "already" scheduled. So user should not be able to control the alignment.

Reply:
Ok thanks.